frontend/src/pages/MilestonesEditor.tsx

```diff
@@ -78,13 +83,43 @@ export const MilestonesEditor = () => {
     setRoadmapsVersionsLocal(roadmapsVersions);
   }, [roadmapsVersions]);
 
+  // Checks if tasks in the tasks list have been given ratings, returns only tasks with ratings from everyone involved
+  const checkRatings = useCallback(
+    (uncheckedTasks: Task[]) => {
+      const ratedTasks: Task[] = [];
+      const developers = allUsers?.filter((user) => user.type === 279715);
+      uncheckedTasks.forEach((task) => {
+        const allRatings = task.ratings.map((rating) => rating.createdByUser);
+        const customerRatings = task.ratings
+          .map((rating) => {
+            return rating.forCustomer;
+          })
+          .filter((value) => value !== null);
+
+        const unratedByCustomer = customers?.filter(
+          (customer) => !customerRatings.includes(customer.id),
+        );
+        const unratedByDeveloper = developers?.filter(
+          (developer) => !allRatings.includes(developer.id),
+        );
+
+        if (unratedByCustomer?.length === 0 && unratedByDeveloper?.length === 0)
+          ratedTasks.push(task);
```

Comment:
```suggestion
        const unratedByDeveloper = developers?.some(
          (developer) => !allRatings.includes(developer.id),
        );

        if (unratedByCustomer?.length === 0 && unratedByDeveloper === false)
          ratedTasks.push(task);
```

Ei tarvitse välttämättä kaikkia käydä läpi, tällä saa saman logiikan mutta short circuiting versiona

Reply:
Nyt muutoksen jälkeen `!unratedByDeveloper` eroaa `unratedByDeveloper === false` (ja aikaisemmasta `unratedByDeveloper?.length === 0`) ehdosta, jos `developers === undefined`.

Jos tämä muutos on muun logiikan puolesta okei, niin 👍

Reply:
Kävin testaamassa ja logiikka toimi edelleen uudella muutoksella myös kuten toivottu 👍